server/db/mysql/adapter.go

```diff
@@ -156,8 +186,12 @@ func (a *adapter) GetDbVersion() (int, error) {
 		return a.version, nil
 	}
 
+	ctx, cancel := a.getContext()
```

Comment:
The purpose of `cancel` is to cancel the query asynchronously, i.e. to launch the query from one go routine then cancel it from another. I don't think we will ever make such async DB calls consequently `cancel` is unlikely to be used ever. If it does not do anything now and unlikely to be used in the future it would make sense to get rid of it completely and remove all the associated boilerplate code.

This combined with the single timeout I mentioned above would allow calls like `a.db.GetContext(a.ctx(), ...)`.

Reply:
OK, so it's an optimization which allows to free resources explicitly sooner, without waiting for the GC. OK.